src/main/java/com/example/napolinights/controller/CheckoutController.java

```diff
@@ -68,37 +75,60 @@ private void initialize() {
         });
     }
 
-    /* ===============================================
-     * SECTION 2: Receiving and Displaying Cart Data
-     * =============================================== */
+    /**
+     * Sets the order ID for this checkout process.
+     * @param orderID The ID of the order to be retrieved.
+     */
+    public void setOrderID(int orderID) {
+        this.orderID = orderID;
+        loadOrderDetails(); // Load order details using the orderID
+    }
 
     /**
-     * Receives data from the OrderController and populates the order summary table.
-     * @param items The array of cart items to be displayed in the order summary.
+     * Loads the order details from the database using the orderID.
      */
-    public void receiveData(CartItem[] items) {
-        // Clear the table first
-        orderSummaryTable.getItems().clear();
+    private void loadOrderDetails() {
+        try {
+            Connection connection = SqliteConnection.getInstance(); // Get a connection to the database
+            OrderDAO orderDAO = new OrderDAO(connection);
+            savedOrder = orderDAO.getOrderById(orderID);
+            if (savedOrder != null) {
+                List<OrderItem> fetchedOrderItems = orderDAO.getOrderItemsByOrderId(orderID); // Fetch order items
+                orderItems.setAll(fetchedOrderItems); // Populate observable list
+
+                // Set the items in the order summary table
+                orderSummaryTable.setItems(orderItems);
+
+                // Calculate and update total price
+                updateTotalPrice();
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
 
-        // Add all the items to the table
-        cartItems.addAll(items);
-        orderSummaryTable.setItems(cartItems);
 
-        // Calculate and update total price
-        updateTotalPrice();
-    }
+
+    /* ===============================================
+     * SECTION 2: Displaying Cart Data
+     * =============================================== */
 
     /**
      * Calculates and updates the total price label.
      */
     private void updateTotalPrice() {
         double total = 0;
-        for (CartItem item : cartItems) {
-            total += item.getTotalInc();
+        for (OrderItem item : orderItems) {
+            double itemTotal = item.getItemPrice() * item.getQuantity();
+            double itemGST = itemTotal * 0.1; // Assuming GST is 10%
```

Comment:
The GST rate is hardcoded as 0.1. It would be better to define it as a constant, so it's easier to modify in the future if needed.